pkg/operator/starter.go

```diff
@@ -354,3 +447,154 @@ func withCustomTags(infraLister v1.InfrastructureLister) deploymentcontroller.De
 		return nil
 	}
 }
+
+func assetWithNamespaceFunc(namespace string) resourceapply.AssetFunc {
+	return func(name string) ([]byte, error) {
+		content, err := assets.ReadFile(name)
+		if err != nil {
+			panic(err)
+		}
+		return bytes.ReplaceAll(content, []byte("${NAMESPACE}"), []byte(namespace)), nil
+	}
+}
+
+func withNamespaceDeploymentHook(namespace string) dc.DeploymentHookFunc {
+	return func(_ *opv1.OperatorSpec, deployment *appsv1.Deployment) error {
+		deployment.Namespace = namespace
+		return nil
+	}
+}
+
+func withHypershiftReplicasHook(isHypershift bool, guestNodeLister corev1listers.NodeLister) dc.DeploymentHookFunc {
+	if !isHypershift {
+		return csidrivercontrollerservicecontroller.WithReplicasHook(guestNodeLister)
+	}
+	return func(_ *opv1.OperatorSpec, deployment *appsv1.Deployment) error {
+		// TODO: get this information from HostedControlPlane.Spec.AvailabilityPolicy
+		replicas := int32(1)
+		deployment.Spec.Replicas = &replicas
+		return nil
+	}
+
+}
+
+func withHypershiftDeploymentHook(isHypershift bool, hypershiftImage string) dc.DeploymentHookFunc {
```

Comment:
(not sure where to put this comment)
In HyperShift, Secrets with cloud credentials is `cloud-controller-creds`, not `ebs-cloud-credentials`. We could either convince HyperShift to create a new IAM role for the CSI driver (with perms. to manipulate only storage) +  CPO could create Secrets for it, or aws-ebs-csi-driver-operator must use the common secret `cloud-controller-creds` used by KCM and everything else.

Reply:
`cloud-controllers-creds` is not meant to be used by storage, the associated IAM role doesn't have the required permissions to manipulate storage resources.

https://github.com/openshift/hypershift/pull/1751 creates a `ebs-cloud-credentials` secret with a proper IAM role for that.